lib/ansible/plugins/vars/recursive_group_vars.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    vars: recursive_group_vars
+    version_added: "?"
+    short_description: Loads group_vars recursively
+    description:
+        - In comparison to host_group_vars plugin, this plugin traverses the `r_group_vars` directory and instead of matching all files and directories within the first matching directory it traverses the directory tree and recursively matches directories and files during directory traversal to groups.
+        - In a directory first matching files are applied and then matching directories traversed. The existing priorities for groups apply for both files and directories (see [How variables are merged](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#how-variables-are-merged)).
+        - If several matching directories are found on the same level they are traversed consecutively in a depth-first fashion.
+        - The deepest found matching file is applied last.
+        - Only files that match a group name are applied. Any other files are ignored.
+        - The following points are identical to host_group_vars plugin:
+            - Files are restricted by extension to one of .yaml, .json, .yml or no extension.
+            - Hidden (starting with '.') and backup (ending with '~') files and directories are ignored.
+            - Only applies to inventory sources that are existing paths.
+    options:
+      _valid_extensions:
+        default: [".yml", ".yaml", ".json"]
+        description:
+          - "Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these."
+          - 'This affects vars_files, include_vars, inventory and vars plugins among others.'
+        env:
+          - name: ANSIBLE_YAML_FILENAME_EXT
+        ini:
+          - section: yaml_valid_extensions
+            key: defaults
+        type: list
+'''
+
+import os
+from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.errors import AnsibleParserError
+from ansible.plugins.vars import BaseVarsPlugin
+from ansible.inventory.group import Group
+from ansible.utils.vars import combine_vars
+
+FOUND = {}
+
+class VarsModule(BaseVarsPlugin):
+
+    def find_vars_recurse(self, loader, groups, path):
+        data = {}
+
+        self._display.debug("recursive_group_vars - Traversing dir : " + path + " with groups : " + to_text(groups))
+
+        # Apply matching files in path
+        for group in groups:
+            try:
+                found_files = loader.find_vars_files(path, group.name, allow_dir=False)
```

Comment:
With `groups` below changing to just be a list of strings, this should be changed

```suggestion
                found_files = loader.find_vars_files(path, group, allow_dir=False)
```